File: cpp/sort-the-people.cpp
#include <algorithm>
#include <unordered_map>
#include "string"
#include "math.h"
#include "vector"

using namespace std;

class Solution {
public:
  vector<string> sortPeople(vector<string>& names, vector<int>& heights) {
    unordered_map<int, int> heightMap;
    vector<string> result;

    for (int i = 0; i < heights.size(); i++) {
      heightMap[heights[i]] = i;
    }
    sort(heights.begin(), heights.end(), greater<int>());
    for (int i = 0; i < heights.size(); i++) {
      result.push_back(names[heightMap[heights[i]]]);
    }

    return result;
  }
};

int main() {
  Solution s;

  return 0;
}

File: java/sort-the-people.java
class Solution {
  public String[] sortPeople(String[] names, int[] heights) {
    int n = names.length;
    String[] res = new String[n];
    List<int[]> list = new ArrayList<>();
    for (int i = 0; i < n; i++) {
      list.add(new int[]{heights[i], i});
    }
    Collections.sort(list, (a, b) -> {
      if (a[0] == b[0]) {
        return names[a[1]].compareTo(names[b[1]]);
      }
      return b[0] - a[0];
    });
    for (int i = 0; i < n; i++) {
      res[i] = names[list.get(i)[1]];
    }
    return res;      
  }
}
